File: app/src/types/visit.ts
export interface VisitRecord {
  id?: number;
  himeId: number;
  visitDate: string; // ISO 8601 date
  memo: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface VisitRecordWithHime extends VisitRecord {
  hime: {
    id: number;
    name: string;
    photoUrl: string | null;
  };
}

export interface VisitFormData {
  himeId: number;
  visitDate: string;
  memo?: string;
}

File: server/internal/models/visit.go
package models

import (
	"time"
)

// VisitRecord 来店記録
type VisitRecord struct {
	ID        uint       `gorm:"primaryKey" json:"id"`
	UserID    uint       `gorm:"not null;index" json:"userId"`
	HimeID    uint       `gorm:"not null;index" json:"himeId"`
	VisitDate time.Time  `gorm:"not null;index" json:"visitDate"`
	Memo      *string    `json:"memo"`
	CreatedAt time.Time  `json:"createdAt"`
	UpdatedAt time.Time  `json:"updatedAt"`
	DeletedAt *time.Time `gorm:"index" json:"-"`

	// リレーション
	User *User `gorm:"foreignKey:UserID" json:"-"`
	Hime *Hime `gorm:"foreignKey:HimeID" json:"hime,omitempty"`
}

// TableName テーブル名を指定
func (VisitRecord) TableName() string {
	return "visit_record"
}
